### PR TITLE
Merge ecs node18 upgrade onto main branch

### PR DESCRIPTION
**Pull request by dlloyd1-ch**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2350

### Change description
The Node 18 upgrade has been isolated on a feature branch to keep it separate from the other release work being done on the main branch. This ensures that the Node 18 upgrade work can be released independently of other work. Once the Node 18 upgrade has gone live, the changes will need to be merged onto the 'main' branch so all future development will be done on Node 18 and ECS.

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by dlloyd1-ch** [comment]
Note: Not to be merged until Node 18 are live.

**Review by mwestacott — COMMENTED** [review]
Looks fine but not approving yet as these two PRs need to be merged in first:

https://github.com/companieshouse/overseas-entities-web/pull/1054
https://github.com/companieshouse/overseas-entities-web/pull/1055

File: ecs-image-build/docker_start.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (AndreKramer7654) [inline ecs-image-build/docker_start.sh:1]: this is just ../docker_start.sh but without the npm i 
could they be combined into one?
if [ $1 = "install" ]; then
  npm i
fi

File: terraform/groups/ecs-service/locals.tf [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    { "name": "PIWIK_START_GOAL_ID", "value": "${var.piwik_start_goal_id}" },
+    { "name": "PIWIK_UPDATE_START_GOAL_ID", "value": "${var.piwik_update_start_goal_id}" },
+    { "name": "SHOW_SERVICE_OFFLINE_PAGE", "value": "${var.show_service_offline_page}" },
+    { "name": "FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION_05072023", "value": "${var.feature_flag_enable_update_statement_validation_05072023}" },
+    { "name": "FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022", "value": "${var.feature_flag_enable_save_and_resume_17102022}" },
+    { "name": "FEATURE_FLAG_ENABLE_ROE_UPDATE_24112022", "value": "${var.feature_flag_enable_roe_update_24112022}" },
+    { "name": "FEATURE_FLAG_ENABLE_ROE_REMOVE_24112022", "value": "${var.feature_flag_enable_roe_remove_24112022}" },
+    { "name": "FEATURE_FLAG_ENABLE_TRUSTS_WEB_07112022", "value": "${var.feature_flag_enable_trusts_web_07112022}" },
+    { "name": "FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME_07032023", "value": "${var.feature_flag_enable_update_save_and_resume_07032023}" },
+    { "name": "FEATURE_FLAG_ENABLE_UPDATE_TRUSTS_30062023", "value": "${var.feature_flag_enable_update_trusts_30062023}" },
```
Comment (AndreKramer7654) [inline terraform/groups/ecs-service/locals.tf:79]: { "name": "FEATURE_FLAG_ENABLE_UPDATE_MANAGE_TRUSTS_29082023", "value": "${var. feature_flag_enable_update_manage_trusts_29082023}" },

Comment (AndreKramer7654) [reply terraform/groups/ecs-service/locals.tf:79]: { "name": "FEATURE_FLAG_DISABLE_UPDATE_PRIVATE_DATA_FETCH_2806202", "value": "${var. feature_flag_disable_update_private_data_fetch_2806202}" },

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+piwik_start_goal_id = "20"
+piwik_update_start_goal_id = "23"
+show_service_offline_page = "false"
+feature_flag_enable_update_statement_validation_05072023 = "false"
+feature_flag_enable_save_and_resume_17102022 = "true"
+feature_flag_enable_roe_update_24112022 = "true"
+feature_flag_enable_roe_remove_24112022 = "false"
+feature_flag_enable_trusts_web_07112022 = "true"
+feature_flag_enable_update_save_and_resume_07032023 = "true"
+feature_flag_enable_update_trusts_30062023 = "false"
```
Comment (AndreKramer7654) [inline terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:20]: feature_flag_enable_update_manage_trusts_29082023 = "false"
feature_flag_disable_update_private_data_fetch_2806202 = "false"

Comment (AndreKramer7654) [reply terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:20]: but these probably will come from separate PRs

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+chs_url = "https://staging.company-information.service.gov.uk"
+piwik_url = "https://matomo.platform.aws.chdev.org/"
+piwik_site_id = "15"
+piwik_start_goal_id = "23"
+piwik_update_start_goal_id = "26"
+cdn_host = "d6nh3dxv55e16.cloudfront.net"
+cookie_domain = "company-information.service.gov.uk"
+account_local_url = "https://identity-staging.company-information.service.gov.uk"
+show_service_offline_page = "false"
+feature_flag_enable_update_statement_validation_05072023 = "false"
```
Comment (AndreKramer7654) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:17]: feature_flag_enable_update_manage_trusts_29082023 = "false"
feature_flag_disable_update_private_data_fetch_2806202 = "false"

Comment (AndreKramer7654) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:17]: were are adding feature_flag_enable_update_statement_validation_05072023  on line 17 so assuming we just add all.

File: ecs-image-build/docker_start.sh (hunk 1 above)
Comment (dlloyd1-ch) [reply ecs-image-build/docker_start.sh:1]: This one in used for the ECS image whereas ../docker_Start.sh is used for the local environment. Keeping them separate is the standard being applied to all migrated services.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars (hunk 4 above)
Comment (dlloyd1-ch) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:17]: These are the values that are currently in live. Any new values will be added in a separate PR.

File: terraform/groups/ecs-service/locals.tf (hunk 2 above)
Comment (mwestacott) [reply terraform/groups/ecs-service/locals.tf:79]: Yep, they are in the two config PRs I have done for CiDev and LIVE.

**Comment by dlloyd1-ch** [comment]
PR Analysis is failing as the main is still building with the old pipeline. Will need to merge the pipeline change first.

**Review by tobafolaranmi-ch — APPROVED** [review]
LGTM

**Review by AndreKramer7654 — APPROVED** [review]

**Pull request merged**